tdigest.go

```diff
@@ -86,6 +91,17 @@ func (t *TDigest) AddCentroid(c Centroid) {
 	}
 }
 
+// Merges the supplied digest into this digest. Functionally equivalent to
+// calling t.AddCentroidList(t2.Centroids(nil)), but avoids making an extra
+// copy of the CentroidList.
+func (t *TDigest) Merge(t2 *TDigest) {
+	t2.process()
+
+	for i := range t2.processed {
```

Comment:
I think we should make this a call to AddCentroidList, that way when we sort out the questions in your other PR it will benefit from what ever decision we make there. Additionally this is how the C++ implementation works.